src/Dashboard/Pages/BuySell.tsx

```diff
@@ -1,142 +1,279 @@
-  position: 'absolute' as 'absolute',
-  top: '50%',
-  left: '50%',
-  transform: 'translate(-50%, -50%)',
+  position: "absolute" as "absolute",
+  top: "50%",
+  left: "50%",
+  transform: "translate(-50%, -50%)",
   width: 400,
   bgcolor: "white",
-  border: '2px solid #000',
+  border: "2px solid #000",
   boxShadow: 24,
   p: 4,
 };
 
+
+const buttonStyle = {
+  height: '30px',
+}
+
 export default function BuySell() {
-  const [isShown, setIsShown] = React.useState(false);
-  const [mode, setMode] = React.useState("NONE");
-  const [ticker, setTicker] = React.useState('');
+  const [isPreviewShown, setIsPreviewShown] = React.useState(false);
+  const [isErrorModalShown, setIsErrorModalShown] = React.useState(false);
+  const [isModalShown, setIsModalShown] = React.useState(false);
+  const [errorModalMessage, setErrorModalMessage] = React.useState("");
+  const [mode, setMode] = React.useState("");
+  const [ticker, setTicker] = React.useState("");
+  const [tickerError, setTickerError] = React.useState(false);
   const [numStocks, setStockAmount] = React.useState(0);
   const [limitPrice, setLimitPrice] = React.useState(0);
-  let marketPrice = 0;
 
+  let marketPrice = 0;
 
   const handleBuyClick = () => {
-    setMode("BUY");
+    setMode("buy");
+
+    var buyButton = document.getElementById("buy-button")
+    var sellButton = document.getElementById("sell-button")
+    if (buyButton && sellButton) {
+      buyButton.style.backgroundColor = "blue"
+      sellButton.style.backgroundColor = "gray"
+    }
   };
 
   const handleSellClick = () => {
-    setMode("SELL");
+    setMode("sell");
+
+    var buyButton = document.getElementById("buy-button")
+    var sellButton = document.getElementById("sell-button")
+    if (buyButton && sellButton) {
+      sellButton.style.backgroundColor = "blue"
+      buyButton.style.backgroundColor = "gray"
+    }
   };
 
   const handleTickerChange = (event: { target: { value: React.SetStateAction<string>; }; }) => {
     setTicker(event.target.value);
-    getMarketPrice(ticker).then((r) => {
-      if (typeof r == 'number') {
+
+    setTickerError(false)
+    getMarketPriceHttp(ticker).then((r) => {
+      if (typeof r == "number") {
         marketPrice = r;
       } else {
         marketPrice = 0;
       }
-    })
-  }
+    });
+
+
+  };
 
-  const handleLimitPriceChange = (event: { target: { value: React.SetStateAction<string>; }; }) => {
+  const handleLimitPriceChange = (event: {
+    target: { value: React.SetStateAction<string> };
+  }) => {
     setLimitPrice(Number(event.target.value));
-  }
+  };
+
+  const handlePreviewClick = async () => {
 
-  const handleConfirmClick = () => {
-    if (ticker !== "" && numStocks !== 0 && mode !== "NONE" && limitPrice !== 0) {
-      setIsShown(true);
-    } else {
-      alert("Enter all required fields.")
+    if (ticker !== "" && numStocks !== 0 && limitPrice !== 0) {
+      setIsPreviewShown(true);
     }
+  };
 
+  const handleReset = () => {
```

Comment:
Do we use this?  Seems like it's not being called.

I think it would be useful to reset all content after a successful API call though